lang/rust/avro/src/schema.rs

```diff
@@ -503,12 +503,12 @@ pub struct UnionSchema {
     // schema index given a value.
     // **NOTE** that this approach does not work for named types, and will have to be modified
     // to support that. A simple solution is to also keep a mapping of the names used.
-    variant_index: HashMap<SchemaKind, usize>,
+    variant_index: BTreeMap<SchemaKind, usize>,
```

Comment:
I've changed the type of the map because otherwise it would be harder to assert the logged error message when UnionSchema validation fails. HashMap's Display impl prints the entries in random order every time.